CDS/scripts/png_to_jpeg.rb

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env ruby
+
+#This module tests if an image file is a PNG and, if it is, attempts to convert it into a JPEG.
+#It depends on ImageMagick being available on the local node (apt-get install ImageMagick or yum install ImageMagick or port install ImageMagick etc.), specifically the 'convert' and 'identify' commands.
+#
+#Arguments:
+# <input_key>{meta:FileNameKey} - use the provided key as an input
+# <output_key>keyname - output the path of the file into this key in the meta: section of the datastore
+
+#END DOC
+
+require 'CDS/Datastore'
+
+#START MAIN
+store=Datastore.new('png_to_jpeg')
+
+input_image=store.substitute_string(ENV['input_key'])
+
+cmdline="identify -format '%m' '#{input_image}'"
+
+command_result = system(cmdline)
+
+if (command_result)
+  cmdline="convert '#{input_image}' '#{input_image}'"
+  system(cmdline)
+else
+  puts "The image is already a JPEG."
```

Comment:
is it possible to trigger this branch if the image is something other than a PNG _or_ a JPEG?

Reply:
Yes. The script it only designed to be used to process JPEGs or PNGs hence the wording.